src/tool_cb_hdr.c

```diff
@@ -104,23 +104,31 @@ size_t tool_header_cb(char *ptr, size_t size, size_t nmemb, void *userdata)
   if(per->config->etag_save_file && etag_save->stream) {
     /* match only header that start with etag (case insensitive) */
     if(curl_strnequal(str, "etag:", 5)) {
-      char *etag_h = NULL;
+      char *etag_v = NULL;
       char *first = NULL;
       char *last = NULL;
       size_t etag_length = 0;
 
-      etag_h = ptr;
+      /*
+       * move etag pointer to header's value
+       * we have to go to the semicolon and trim all the LWS preceding
+       * the field value https://tools.ietf.org/html/rfc2616#section-4.2
+       */
+      etag_v = memchr(ptr, ':', cb);
```

Comment:
You know the colon is at index 4 since the string matched "etag:" already. No need to memchr it...

Reply:
Fixed